File: Srujan-Unbxd-2021-interns test.py
import requests
import json
import csv
ans=[]
def convert(response):
	convert_json = response.json()
	only_products = convert_json["response"]["products"]
	for i in range(len(only_products)):
		for j in only_products[i]:
			if(type(only_products[i][j])==list):
				if(j!="unbxd_color_for_category"):
					only_products[i][j]=list(set(only_products[i][j]))
					if(len(only_products[i][j]) > 1):
						only_products[i][j]=','.join(only_products[i][j])
					else:
						only_products[i][j]=str(only_products[i][j])
				else:
					arr=[]
					for m in only_products[i][j]:
						l=m.split("::")
						arr.append(l[0])
					h=list(set(arr))
					only_products[i][j]=','.join(h)
	return only_products
for start in range(1,10):
	response = requests.get("https://search.unbxd.io/fb853e3332f2645fac9d71dc63e09ec1/demo-unbxd700181503576558/search?&q=*&rows=10&start="+str(start))
	ans.append(convert(response))
file = open('Unbxd-2021-interns test.csv', 'a+', newline ='') 
with file:     
    write = csv.writer(file) 
    write.writerows(ans)

File: Srujan-Unbxd-2021-interns test.java
import java.net.HttpURLConnection;
import java.net.URL;
import java.util.Scanner;
import org.json.simple.JSONArray;
import org.json.simple.JSONObject;
import org.json.simple.parser.JSONParser;

class test
{
	public static void main(String [] args)
	{
		//inline will store the JSON data streamed in string format
		String inline = "";
		inline.split(",") // splitting each comma separated value
		for(int i=0;i<String.size();i++)
		{
		try
		{
			//Sending url for each iteration
			URL url = new URL("https://search.unbxd.io/fb853e3332f2645fac9d71dc63e09ec1/demo-unbxd700181503576558/search?&q=*&rows=i&start=i");
			HttpURLConnection conn = (HttpURLConnection)url.openConnection();
			conn.setRequestMethod("GET");
			conn.connect();
			int responsecode = conn.getResponseCode();
			if(responsecode != 200)
				throw new RuntimeException("HttpResponseCode: " +responsecode);
			else
			{
				//Scanner functionality will read the JSON data from the stream
				Scanner sc = new Scanner(url.openStream());
				while(sc.hasNext())
				{
					inline+=sc.nextLine();
				}
				System.out.println(inline);
				sc.close();
			}
			JSONParser parse = new JSONParser();
			JSONObject jobj = (JSONObject)parse.parse(inline);
			JSONArray jsonarr_1 = (JSONArray) jobj.get("searchMetaData");
			for(int i=0;i<jsonarr_1.size();i++)
			{
				JSONObject jsonobj_1 = (JSONObject)jsonarr_1.get(i);
				JSONArray jsonarr_2 = (JSONArray) jsonobj_1.get("products");;
				System.out.println("Elements under products array");
				for(int j=0;j<jsonarr_2.size();j++)
				{
					JSONObject jsonobj_2 = (JSONObject) jsonarr_2.get(j);
					System.out.println("\n");
				}
			}
			conn.disconnect();
		}
		}
		catch(Exception e)
		{
			e.printStackTrace();
		}
	}